lib/sycamore/sycamore/transforms/llm_query.py

```diff
@@ -52,28 +54,63 @@ def __init__(
         self._per_element = per_element
         self._format_kwargs = format_kwargs
         self._number_of_elements = number_of_elements
+        self._element_type = element_type
 
     def execute_query(self, document: Document) -> Document:
-        if self._per_element:
-            elements = document.elements
+        final_prompt = self._prompt
+        element_count = 0
+        if self._per_element or self._number_of_elements: 
             for idx, element in enumerate(document.elements):
-                elements[idx] = self._query_text_object(element)
-                if self._number_of_elements and idx >= self._number_of_elements:
-                    break
-            document.elements = elements
-        elif self._number_of_elements:  # limit to a number of elements
-            text_representation = self._prompt
-            for idx, element in enumerate(document.elements):
-                text_representation += "\n" + element["text_representation"]
-                if idx >= self._number_of_elements:
-                    break
-            prompt_kwargs = {"prompt": text_representation}
-            llm_resp = self._llm.generate(prompt_kwargs=prompt_kwargs, llm_kwargs=self._llm_kwargs)
-            document["properties"][self._output_property] = llm_resp
+                if self._element_type and element.type != self._element_type:
```

Comment:
You need to add the `else` block here. Otherwise you are not doing anything with the `element_type`

Reply:
Nvm I see the rationale